atomicapp/nulecule/lib.py

```diff
@@ -94,7 +101,11 @@ def get_provider(self, provider_key=None, dry=False):
         # If provider_key isn't provided via CLI, let's grab it the configuration
         if provider_key is None:
             provider_key = self.config.get(GLOBAL_CONF)[PROVIDER_KEY]
-        provider_class = self.plugin.getProvider(provider_key)
+        try:
+            provider_class = self.plugin.getProvider(provider_key)
+        except NuleculeException as e:
+            logger.error(e)
+            sys.exit(1)
```

Comment:
I don't really know of another place we exit directly from the code like this rather than raising an exception. I'll have to ask the other guys for review.